### ISSUE TITLE
Division operation will fail when mask.to(input).sum(dim=1) equals zero. This will result in division by zero, causing NaN values in the loss.

### ISSUE THREAD

**Issue by slowmoyang** [nurion4hep/DeepMuonReco#13]
very low priority

Division operation will fail when mask.to(input).sum(dim=1) equals zero. This will result in division by zero, causing NaN values in the loss.
```suggestion
            denom = mask.to(input).sum(dim=1).clamp(min=1)
            loss = loss.sum() / denom
```

_Originally posted by @Copilot in https://github.com/nurion4hep/DeepMuonReco/pull/9#discussion_r2351981635_
